File: shoppingm8_fe/lib/list/invitation/dto/listInvitationDto.dart
import 'package:shoppingm8_fe/list/dto/listResponseDto.dart';
import 'package:shoppingm8_fe/user/dto/userDto.dart';

class ListInvitationDto {
  final num id;
  final UserDto invited;
  final UserDto inviting;
  final ListResponseDto list;

  ListInvitationDto({this.id, this.invited, this.inviting, this.list});

  factory ListInvitationDto.fromJson(Map<String, dynamic> json) {
    return ListInvitationDto(
      id: json['id'],
      invited: UserDto.fromJson(json['invited']),
      inviting: UserDto.fromJson(json['inviting']),
      list: ListResponseDto.fromJson(json['list']),
    );
  }
}

File: ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/invitation/api/dto/ListInvitationDto.java
package edu.pw.shoppingm8.list.invitation.api.dto;

import edu.pw.shoppingm8.list.api.dto.ListDto;
import edu.pw.shoppingm8.list.invitation.db.ListInvitation;
import edu.pw.shoppingm8.user.api.dto.UserDto;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class ListInvitationDto {
    Long id;
    UserDto invited;
    UserDto inviting;
    ListDto list;

    public static ListInvitationDto of(ListInvitation entity) {
        return ListInvitationDto.builder()
                .id(entity.getId())
                .invited(UserDto.of(entity.getInvited()))
                .inviting(UserDto.of(entity.getInviting()))
                .list(ListDto.of(entity.getList()))
                .build();
    }
}
